### PR TITLE
CopilotChat: Replace "SK" with "SemanticKernel" and add App Service Plan allowable values.

### PR DESCRIPTION
**Pull request by adrianwyatt**
### Motivation and Context
Remove most occurrences of "SK" from CopilotChat deployment

### Description
- Renamed most occurrences of "SK" to either "SemanticKernel" or "SemKer" throughout CopilotChat
- Updated App Plan SKUs with a list of allowed values.

### CODE REVIEW

**Comment by adrianwyatt** [comment]
Still doing some validation - hold merge until I've finished.

**Review by hathind-ms — DISMISSED** [review]

**Comment by hathind-ms** [comment]
This change is missing: 

Package URI from https://skaasdeploy.blob.core.windows.net/api/skaas.zip to be https://semantic-kernel-deploy.blob.core.windows.net/api/Semantic-Kernel.zip

**Comment by adrianwyatt** [comment]
> This change is missing:
> 
> Package URI from https://skaasdeploy.blob.core.windows.net/api/skaas.zip to be https://semantic-kernel-deploy.blob.core.windows.net/api/Semantic-Kernel.zip

We'll need to standup another blob service and populate it. I wanted to not block these changes on that move. I'll look at that next.

**Review by hathind-ms — DISMISSED** [review]

File: samples/apps/copilot-chat-app/webapi/DeploymentTemplates/sk-existing-azureopenai.bicep [hunk 1]
```diff
@@ -6,9 +6,10 @@ Bicep template for deploying Semantic Kernel to Azure as a web app service with
 */
 
 @description('Name for the deployment')
```
Comment (glahaye) [inline samples/apps/copilot-chat-app/webapi/DeploymentTemplates/sk-existing-azureopenai.bicep:8]: Suggestion from Nilesh:

```suggestion
@description('Name for the deployment - Must consist of alphanumeric characters or \'-\'')
```

File: samples/apps/copilot-chat-app/webapi/DeploymentTemplates/sk-existing-openai.bicep [hunk 2]
```diff
@@ -6,9 +6,10 @@ Bicep template for deploying Semantic Kernel to Azure as a web app service with
 */
 
 @description('Name for the deployment')
```
Comment (glahaye) [inline samples/apps/copilot-chat-app/webapi/DeploymentTemplates/sk-existing-openai.bicep:8]:
```suggestion
@description('Name for the deployment - Must consist of alphanumeric characters or \'-\'')
```

File: samples/apps/copilot-chat-app/webapi/DeploymentTemplates/sk-new.bicep [hunk 3]
```diff
@@ -6,9 +6,10 @@ Bicep template for deploying Semantic Kernel to Azure as a web app service with
 */
 
 @description('Name for the deployment')
```
Comment (glahaye) [inline samples/apps/copilot-chat-app/webapi/DeploymentTemplates/sk-new.bicep:8]:
```suggestion
@description('Name for the deployment - Must consist of alphanumeric characters or \'-\'')
```

File: samples/apps/copilot-chat-app/webapi/DeploymentTemplates/main.bicep [hunk 4]
```diff
@@ -9,9 +9,10 @@ Resources to add:
 */
 
 @description('Name for the deployment')
```
Comment (glahaye) [inline samples/apps/copilot-chat-app/webapi/DeploymentTemplates/main.bicep:11]:
```suggestion
@description('Name for the deployment - Must consist of alphanumeric characters or \'-\'')
```

File: samples/apps/copilot-chat-app/webapi/DeploymentTemplates/main.bicep [hunk 5]
```diff
@@ -42,7 +43,7 @@ param endpoint string = ''
 param apiKey string = ''
 
 @description('Semantic Kernel server API key - Provide empty string to disable API key auth')
```
Comment (glahaye) [inline samples/apps/copilot-chat-app/webapi/DeploymentTemplates/main.bicep:45]: Other suggestion from Nilesh (and again in all 4 bicep files:

```suggestion
@description('Semantic Kernel server API key - Generated GUID by default\nProvide empty string to disable API key auth')
```

**Review by hathind-ms — APPROVED** [review]

**Review by hathind-ms — APPROVED** [review]

**Review by glahaye — APPROVED** [review]

**Pull request merged**